### ISSUE TITLE
Collecting static directories into one directory

### ISSUE THREAD

**Issue by okaycj** [lookit/lookit-api#852]
**TL;DR**: All static files should be under the same parent directory. 

**Narrative**
When generating the integrity hash for some JS files, I found static files in many different locations.   It would be nicer and easier to find the files if they were under the same static directory.  

**Acceptance Criteria**
- [x] All static files are under the static directory
- [x] `collectstatic` works correctly on development

**Implementation Notes**
Moving the files and renaming their location within the templates seems straight forward.  I am unsure how GCP will react to change in static file location.

**Comment by becky-gilbert** [comment]
## Background

I did some more reading on this and it turns out that, by default, Django expects static files to be stored at the _app-level_ at `<APP_NAME>/static`. We don't necessarily need to follow this convention, but the `collectstatic` command will automatically look for static files in a `static` subdirectory of each installed app. 

When following this convention, it's good practice to add a subfolder with the app name inside of each app's static directory (e.g. `my_app/static/my_app/js`), to avoid file name collisions across apps. 

We can use `STATICFILES_DIRS` to specify other locations for `collectstatic` to look in when collecting files.

## Current state

We are currently using this organizational convention for our apps (though two of them don't contain any static files, and one doesn't use the app name subfolder):

- accounts: no static files
- api: no static files
- exp: `exp/static/exp`
- studies: `studies/static/studies`
- web: `web/static` (no `web` subfolder)

We've also added the `project/_static` directory to the set of locations that `collectstatic` will look for files:
```
STATICFILES_DIRS = (os.path.join(BASE_DIR, "_static"),)
```

The reason this additional folder has an underscore is because the `project/static` folder name is already used as the location where `collectstatic` will copy all of the files it finds:
```
STATIC_ROOT = os.path.join(BASE_DIR, "static")
```
so we need to avoid a name collision (which isn't obvious because this folder doesn't exist until `collectstatic` runs).

## Problems

1. There's still a possibility of file name collisions because (a) the web app doesn't have a `web` subfolder in `static` AND (b) we've added the additional `_static` location with no unique subdirectory.

2. Even if the convention is to have separate static directories for each app (plus an extra one), it might not be intuitive for us, and we might prefer to have all static files in one place.

## Options for reorganizing:

1. Leave things the way they are, except add a `web` subfolder in `web/static`. This fixes problem 1 above, but not 2.
3. Move everything to `_static`. This fixes both problems above, but is unconventional. 
4. Something else??

## Sources
 
https://docs.djangoproject.com/en/4.1/howto/static-files/
https://testdriven.io/blog/django-static-files/
https://www.digitalocean.com/community/tutorials/working-with-django-templates-static-files

p.s. to make things extra confusing, we have one other static folder here: `exp/tests/static`, but this folder might've been separated intentionally if we don't want/need these testing-specific files to be collected via `collectstatic`?

**Comment by mekline** [comment]
Thanks Becky!  Some questions to help me understand what's going on: 

- Is anything in project/_static/ right now, or has it just been added as a potential place we might put static files?   
- Is exp/tests/static the right place for test files? Are there equivalents in web and studies? What would the downside be to having testing-specific files collected in the collectstatic process?
- If you keep the static files in a similar (more standardized) set of locations as they are now, will you still be able to make some progress in reducing duplication/factoring/templating as described in #853 ?

**Comment by becky-gilbert** [comment]
Hi Melissa! CJ and I just had a chat about this and settled on moving everything into `web/static` (option 3!), if that's ok with you.

> Is anything in project/_static/ right now, or has it just been added as a potential place we might put static files?

There's only a few things there (some of which we could get from a CDN instead of hosting ourselves). Very easy to move.

I'd appreciate @okaycj 's input on your last two questions since I don't know for sure. But in any case, here are my thoughts:

> Is exp/tests/static the right place for test files? Are there equivalents in web and studies? What would the downside be to having testing-specific files collected in the collectstatic process?

I don't know what the "right"/conventional way to do this is, but I can do some research. The only reason I can think of for keeping the test static files separate is to reduce clutter and storage costs related to hosting files that aren't actually used on staging or production. The reason for putting in the same place as everything else is for 

> If you keep the static files in a similar (more standardized) set of locations as they are now, will you still be able to make some progress in reducing duplication/factoring/templating as described in https://github.com/lookit/lookit-api/issues/853 ?

I think this decision shouldn't have any bearing on the templating issue. It should just require changing file locations in the templates, which isn't a big deal.

**Comment by becky-gilbert** [comment]
I've opened PR #1101 in which I've done the following:

- Moved all static files from `project/_static`, `exp/static` and `studies/static` -> `web/static`
- Deleted `project/_static`, `exp/static` and `studies/static` 
- Removed `project/_static` as an additional static file directory (i.e. STATICFILES_DIRS in settings.py)
- Updated relevant static file paths in template files
- Added `collectstatic` to Make file for convenience
- Manually checked (locally) all templates that contained file path changes

So the static file organization now looks like this:
- web
  - static
    - css
       - font
     - fonts
     - images
     - js
     - videos

After running `collectstatic`, another static folder will appear in project (`project/static`). This folder contains everything from `web/static` plus a bunch of other stuff (specifically: ace_overlay, admin, django_extensions, guardian, rest_framework).

To do: 
- [ ] Fix errors on participant analytics and participant contact pages (console errors: none of the SHA hashes match, moment not defined).
- [ ] Test on staging to make sure `collectstatic` and cloud storage locations are all still working.
- [ ] Decide whether or not to move the static files in `exp/tests/static`.

Notes:
- `bootstrap-editable.css` references `../img/loading.gif` and `../img/clear.png`. I would've changed these file locations (`img` -> `images`) but I can't find either file (though `loading.gif` does exist in the EFP repo). 
- I would've put all font files in `web/static/fonts`, but `summernote.min.css` references font files from a font subfolder and editing it would be bad practice.
- There are set of 4 font files (`fonts/glyphicons-halflings-regular`) that don't seem to be used anywhere. Remove?

**Comment by becky-gilbert** [comment]
@okaycj not sure if you saw my comment above? Just wondering whether you have any thoughts, specifically on these things?
- whether or not to move the static files in `exp/tests/static`?
- `exp/static/exp/css/bootstrap-editable.css` references `../img/loading.gif` and `../img/clear.png`, e.g.
     
    ```css
    .editableform-loading {
        background: url('../img/loading.gif') center center no-repeat;  
        height: 25px;
        width: auto; 
        min-width: 25px; 
    }
    ```
    I would've changed these file locations but I can't actually find either image file anywhere in the repo. Am I missing something here? 
- There are set of 4 font files (`fonts/glyphicons-halflings-regular`) that don't seem to be used anywhere. Should I remove them?
